tianshou/policy/modelbase/psrl.py

```diff
@@ -0,0 +1,223 @@
+            (n_state, n_action, n_state).
+        :param np.ndarray rew_sum: total rewards, with shape
+            (n_state, n_action).
+        :param np.ndarray rew_square_sum: total rewards' squares, with shape
+            (n_state, n_action).
+        :param np.ndarray rew_count: the number of rewards, with shape
+            (n_state, n_action).
+        """
+        self.updated = False
+        self.trans_count += trans_count
+        sum_count = self.rew_count + rew_count
+        self.rew_mean = (self.rew_mean * self.rew_count + rew_sum) / sum_count
+        self.rew_square_sum += rew_square_sum
+        raw_std2 = self.rew_square_sum / sum_count - self.rew_mean ** 2
+        self.rew_std = np.sqrt(1 / (
+            sum_count / (raw_std2 + self.__eps) + 1 / self.rew_std_prior ** 2))
+        self.rew_count = sum_count
+
+    @staticmethod
+    def sample_from_prob(trans_count: np.ndarray) -> np.ndarray:
+        sample_prob = np.zeros_like(trans_count)
+        n_s, n_a = trans_count.shape[:2]
+        for i in range(n_s):
+            for j in range(n_a):  # numba does not support dirichlet :(
+                sample_prob[i][j] = np.random.dirichlet(trans_count[i][j])
+        return sample_prob
+
+    def sample_from_rew(self) -> np.ndarray:
+        return np.random.normal(self.rew_mean, self.rew_std)
+
+    def solve_policy(self) -> None:
+        self.updated = True
+        self.policy, self.value = self.value_iteration(
+            self.sample_from_prob(self.trans_count),
+            self.sample_from_rew(),
+            self.discount_factor,
+            self.eps,
+            self.value,
+        )
+
+    @staticmethod
+    def value_iteration(
+        trans_prob: np.ndarray,
+        rew: np.ndarray,
+        discount_factor: float,
+        eps: float,
+        value: np.ndarray,
+    ) -> np.ndarray:
+        """Value iteration solver for MDPs.
+
+        :param np.ndarray trans_prob: transition probabilities, with shape
+            (n_state, n_action, n_state).
+        :param np.ndarray rew: rewards, with shape (n_state, n_action).
+        :param float eps: for precision control.
+        :param float discount_factor: in [0, 1].
+        :param np.ndarray value: the initialize value of value array, with
+            shape (n_state, ).
+
+        :return: the optimal policy with shape (n_state, ).
+        """
+        Q = rew + discount_factor * trans_prob.dot(value)
+        new_value = Q.max(axis=1)
+        while not np.allclose(new_value, value, eps):
+            value = new_value
+            Q = rew + discount_factor * trans_prob.dot(value)
+            new_value = Q.max(axis=1)
+        # this is to make sure if Q(s, a1) == Q(s, a2) -> choose a1/a2 randomly
+        Q += eps * np.random.randn(*Q.shape)
+        return Q.argmax(axis=1), new_value
+
+    def __call__(
+        self,
+        obs: np.ndarray,
+        state: Optional[Any] = None,
+        info: Dict[str, Any] = {},
```

Comment:
These two arguments are not used.

Reply:
That's to be consistent with model API

Reply:
What do you mean by "model API"? PSRLModel is the first model and I don't see it has any base-class.